src/sdk/pynni/nni/compression/torch/speedup/infer_shape.py

```diff
@@ -552,25 +718,44 @@ def relu_inshape(module_masks, mask):
     """
     assert isinstance(mask, CoarseMask)
     if module_masks.input_mask is not None:
-        # check if has a mask conflict
-        assert module_masks.input_mask <= mask
+        # mask conflict should be solved before speedup
+        assert module_masks.input_mask == mask
```

Comment:
From a semantic point of view, I think `<=` maybe better. 
Since currently we pass the mask is by reference, here, `== `and `<=` have no difference, because `mask` and `input_mask` actually point to the same tensor. Both of these assertions are right. But semantically speaking, I recommend using `<=`. Especially when we intend to replace the `mask passing by reference` to `passing by value`, the cat scenario will fail on this assertion if we use `==`:

For example, relu(cat(conv1, conv2)), 
The out_mask of the first visit to `cat` from `conv1` may be {1: 1, 2, 3}, and after the second visit to `cat` from `conv1`, the output mask may have some incremental update, let's say it will be {1: 1, 2, 3, 6,7,8}. and it will fail on the assert.

Reply:
Thanks for good catch, updated.